### PR TITLE
jgrss/crs error

### PR DESCRIPTION
**Pull request by jgrss**
## What is this PR changing?
This PR fixes a bug with unsupported EPSG codes in calls to `to_raster()`. For example, [EPSG:8858](https://epsg.io/8858) cannot be translated to WKT directly by `rasterio.CRS`. Instead, it requires a translation to `pyproj.CRS`, which can then be used to translate to a string by `to_wkt()` or `to_proj4()`.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -34,7 +34,7 @@ install_requires =
     opencv-python
     pandas>=1.*             # via -r requirements.in, dask, geopandas, xarray
     pyproj>=2.6.0             # via -r requirements.in, geopandas
-    rasterio>=1.2.0,<=1.2.6
+    rasterio>=1.3.0,<2.0.0
```
Comment (jgrss) [inline setup.cfg:37]: This supports `stackstac`.

File: setup.cfg [hunk 2]
```diff
@@ -76,7 +76,7 @@ view = ipython
 web = gsutil
     requests
     pystac-client
-    stackstac[viz]
+    stackstac[viz]@git+https://github.com/jgrss/stackstac.git  # contains some pyproj temp fixes
```
Comment (jgrss) [inline setup.cfg:79]: I had to fork and make some minor `pyproj` changes.

File: src/geowombat/core/io.py [hunk 3]
```diff
@@ -769,16 +770,27 @@ def to_raster(
...
-        kwargs['crs'] = data.crs
-
     if 'transform' not in kwargs:
         kwargs['transform'] = data.gw.transform
 
     if 'num_threads' in kwargs:
         if isinstance(kwargs['num_threads'], str):
             kwargs['num_threads'] = 'all_cpus'
 
+    if 'crs' in kwargs:
```
Comment (jgrss) [inline src/geowombat/core/io.py:780]: This is the main change. I.e., translating input CRS codes to `pyproj.CRS` classes that can support any EPSG code.

File: src/geowombat/radiometry/sr.py [hunk 4]
```diff
@@ -95,29 +95,36 @@ def s_atm(r):
 
 
 def _format_coeff(dataframe, sensor, key):
-
-    bands_dict = dict(l5={'1': 'blue', '2': 'green', '3': 'red', '4': 'nir', '5': 'swir1', '6': 'th', '7': 'swir2'},
-                      l7={'1': 'blue', '2': 'green', '3': 'red', '4': 'nir', '5': 'swir1', '6VCID1': 'th1',
-                          '6VCID2': 'th2', '7': 'swir2', '8': 'pan'},
-                      l8={'1': 'coastal', '2': 'blue', '3': 'green', '4': 'red', '5': 'nir', '6': 'swir1',
-                          '7': 'swir2', '8': 'pan', '9': 'cirrus', '10': 'th1', '11': 'th2'})
+    bands_dict = dict(
```
Comment (jgrss) [inline src/geowombat/radiometry/sr.py:98]: This is already merged and didn't get squashed.

File: src/geowombat/radiometry/sr.py [hunk 5]
```diff
@@ -148,12 +150,17 @@ def get_landsat_coefficients(meta_file):
 
                 sensor, m_l, a_l, m_p, a_p, date_acquired, sza
         """
-
-        associations = {'LANDSAT_5': 'l5',
-                        'LANDSAT_7': 'l7',
-                        'LANDSAT_8': 'l8'}
-
-        MetaCoeffs = namedtuple('MetaCoeffs', 'sensor m_l a_l m_p a_p date_acquired sza left bottom right top')
+        associations = {
```
Comment (jgrss) [inline src/geowombat/radiometry/sr.py:153]: Also already merged

**Review by mmann1123 — APPROVED** [review]
Looks good to me. Although we need to start adding tests as we go. Sorry I've been distracted with other work.

**Comment by jgrss** [comment]
> Looks good to me. Although we need to start adding tests as we go. Sorry I've been distracted with other work.

Yeah, good call. I added a few things to this PR, mainly a change from `rasterio.CRS` to `pyproj.CRS` to better support EPSG codes. I also updated a couple of tests.

**Pull request merged**